src/clib/xtg/libxtg.h

```diff
@@ -1864,6 +1864,22 @@ grdcp3d_corners(long ic,
                 long n_swig_np_flt_inplaceflat_v1,
                 double corners[]);
 
+void
+grdcp3d_get_vtk_esg_geometry_data(long ncol,
+                                  long nrow,
+                                  long nlay,
+
+                                  double *swig_np_dbl_inplaceflat_v1,  // coordsv
+                                  long n_swig_np_dbl_inplaceflat_v1,
+                                  float *swig_np_flt_inplaceflat_v1,   // zcornsv
+                                  long n_swig_np_flt_inplaceflat_v1,
+
+                                  double *swig_np_dbl_aout_v1,         // output vertex arr
+                                  long n_swig_np_dbl_aout_v1,          // allocated length of vertex arr
+                                  long *swig_np_lng_aout_v1,           // hex connectivity array (out)
+                                  long n_swig_np_lng_aout_v1,          // allocated length of hex conn arr
+                                  long* swig_lon_out_p1);              // number of vertices actually written into vertex arr
```

Comment:
Would it be better just to return this from the function? I.e. `long grdcp3d_get_vtk_esg_geometry_data( ` instead of `void`

Reply:
I agree and would like to, but I'm not sure how this could be accomplished in the presence of the Numpy/Swig magic that is happening here. 
Do you have any examples of this usage pattern where a simple value is being returned in combination with siwig/numpy arrays?

Reply:
Yes example here:
https://github.com/equinor/xtgeo/blob/ce885e7ba2a9a3927b226a57ee7b70b01239caac/src/clib/xtg/libxtg.h#L226

In test, the int is returned first, then the swig magics:
https://github.com/equinor/xtgeo/blob/ce885e7ba2a9a3927b226a57ee7b70b01239caac/tests/test_common/test_calc.py#L354

Reply:
Thanks for the input. This is now fixed.